include/mitsuba/core/distr_1d.h

```diff
@@ -209,7 +214,26 @@ template <typename Value> struct DiscreteDistribution {
     }
 
 private:
-    void compute_cdf(const ScalarFloat *pmf, size_t size) {
+    void compute_cdf() {
+        static_assert(dr::is_jit_v<Float>,
+                      "This method should only be used in JIT variants!");
+
+        if (m_pmf.empty())
+            Throw("DiscreteDistribution: empty distribution!");
+        if (!dr::all(m_pmf >= 0.f))
+            Throw("DiscreteDistribution: entries must be non-negative!");
+        if (!dr::any(m_pmf > 0.f))
+            Throw("DiscreteDistribution: no probability mass found!");
+
+        m_cdf = dr::prefix_sum(m_pmf, false);
```

Comment:
Previously I did this but it complains about `m_pmf` attached to the AD graph. 
So i detached it before feeding it to prefix

Reply:
The operation is differentiable in the next version of Dr.Jit, but the one on nanobind master raises an exception when given differentiable input to be on the safe side. Detaching is a reasonable workaround for now, and I think we usually in any case don't want to differentiate with respect to the PMF. @ziyi-zhang , it might be worth investigating why `m_pmf` is differentiable in the first place.

Reply:
I haven't added any explicit `dr::detach` statements, just in case someone eventually wants to differentiate through the PMF/PDF. I don't mind adding them, if you feel like the risk of accidentally differentiating w.r.t. the distribution outweighs this _future_ feature.